### ISSUE TITLE
support casting in print expressions

### ISSUE THREAD

**Issue by heycam** [Pernosco/pernosco#22]
I have an Executions window for a method on a Gecko `nsIFrame` subclass. I want to add a print expression to show the document URL.  I tried

```
this->mContent.mRawPtr->mNodeInfo.mRawPtr->mDocument->mDocumentURI.mRawPtr->mSpec.mData
```

but this doesn't work since `mSpec` is a member of `nsStandardURL` and not `nsIURI`.  So I suppose I want:

```
((mozilla::net::nsStandardURL*)(this->mContent.mRawPtr->mNodeInfo.mRawPtr->mDocument->mDocumentURI.mRawPtr))->mSpec.mData
```

but that doesn't work since casting is not supported in the expression syntax.

**Comment by rocallahan** [comment]
Hmm, we do have autodowncasting to the concrete type since recently so the auto cast should have worked.

**Comment by heycam** [comment]
Unfortunately it tells me for `->mSpec` that it's "Not a field of the left-hand-side value".

**Comment by khuey** [comment]
Can you give us a link to a recording and an execution to search for to use that expression with?

**Comment by heycam** [comment]
https://pernos.co/debug/tmxwmjtBWbCETqlQhKmdhQ/index.html and search for `nsImageFrame::OnFrameUpdate`.

**Comment by khuey** [comment]
What you are actually trying to do here will be fixed by (internal) 36f96ffc9dc3652af01d3979a3d3cbef80b59b4a. As filed this is still not fixed, since we don't yet support actual casts, but I'm going to close this anyways.
